### ISSUE TITLE
При отправке нового комментария сбрасывается счетчик непрочитанных 

### ISSUE THREAD

**Issue by nlopin** [vas3k/vas3k.club#332]
## Ситуация
Приходишь в тему почитать комментарии, но когда отвечаешь на один из них, страница перезагружается и все непрочитанные комменты становятся прочитанными.

## Решение
Отправлять комментарии асинхронно на сервер.

Сервер все так же будет обновлять все свои счетчики просмотренных сообщений, но в рамках текущего экрана ничего не сбросится, т.к. не будет ререндера

### Технические детали

Решения, на которых я сделаю реализацию. Если у вас есть идеи, как можно сделать лучше, пишите в комменты:

* сервер должен возвращать html-сниппет комментария, чтобы избежать дублирования шаблонов на фронтенде (сейчас у нас 4 отображения для комментариев)
* форма отправки комментария будет блокироваться до получения ответа сервера. Оптимистичный рендер сделать не выйдет, т.к. клиентский код не знает верстку (следует из предыдущего пункта)

❓ Есть нерешенная проблема вставки. Сервер отдает html, но не всегда понятно, куда его вставлять. Скорее всего придется добавить в верстку какую-то доп.инфу, чтобы это понять

**Comment by vas3k** [comment]
Вот я бы тоже с удовольствием послушал best practices для такого. Фича нужная так-то. 

Отправлять с сервера отрендеренный HTML кажется странным, но дублировать шаблоны комментариев — ещё большее зло. Мы моментально запутаемся где вносить правки и всё проебём.

Мой вариант: придерживаться json api и попробовать как-то аккуратно перенести рендеринг комментариев с сервера на клиент. Не подгружать их асинхронно, а только рендерить, как сейчас сделано с голосованием, например.

Вещи, которые надо не забыть при этом:
- Комментарии рендерятся не только в постах, но и в профиле юзера или в поиске, например. Самый лучший вариант там — сделать для них отдельный шаблон и оставить его на сервере. На них давно жалуются, что они выглядят как комментарии, но не работают как они (кнопка ответить например ничего не делает). Вот и пофиксим :)
- Сам виджет vote теперь тоже можно будет засунуть внутрь клиентского рендеринга, а в остальных местах использовать заглушку, как ты сделал в поиске. Так норм.
- JSON-эндпоинты решили делать через добавление .json к уже имеющимся. Посмотри как сделано получение профиля юзера, там есть сразу и вьюха и вся нужная авторизация.

**Comment by nlopin** [comment]
Мне нравится подход, аналогичный upvote. Не вижу большой проблемы отрендерить и комментарий, и кнопку. 

Я посмотрю, как можно решить историю с шаблонами. В идеале сервер больше ничего о них не должен знать.

Одним PR тут не обойдешься, давай делать кусками:
* перенос рендеринга комментариев на клиент
* новый шаблон для рендера комментариев в профиле пользователя и поиске
* добавление нового комментария

**Comment by vas3k** [comment]
> В идеале сервер больше ничего о них не должен знать.

Да, именно. На сервере теперь можно будет завести один общий шаблон комментария, который мы будем использовать в поиске или профиле юзера. Потом подтюним ему стили и добавим в его блок название поста, а то многие спрашивают как из поиска по комментам перейти в пост. Одного шаблона для этого хватит.

> Одним PR тут не обойдешься, давай делать кусками

Я бы поменял местами 1 и 2. Ведь если ты сразу перенесешь рендеринг на клиент, это автоматом сломает поиск и профиль :)

**Comment by nlopin** [comment]
- [Новый шаблон для отображения комментариев в поиске и профиле. При клике переходить в пост]
- [Перенести рендер комментариев на клиент] 
- [Асинхронная отправка комментариев]

**Comment by nlopin** [comment]
Есть два варианта показать комменты — использовать стандартный блок комментария, либо уменьшенный (ака reply). Я не уверен, что комменты это самое важное в поиске, поэтому показывать их минифицированно мне кажется хорошей идеей.

Ниже сравнение норм vs сжато:

| Норм        | Мини           | 
| ------------- |-------------|
| ![image](https://user-images.githubusercontent.com/1469636/89223342-e92d5000-d5d6-11ea-8b42-4b42a0e2c031.png) | ![CleanShot 2020-08-03 at 22 19 58@2x](https://user-images.githubusercontent.com/1469636/89224036-1af2e680-d5d8-11ea-8531-19ea4e493da9.png) |
|![CleanShot 2020-08-03 at 22 20 30@2x](https://user-images.githubusercontent.com/1469636/89224168-5097cf80-d5d8-11ea-88f4-0a36a2c8d7ce.png) | ![CleanShot 2020-08-03 at 22 19 51@2x](https://user-images.githubusercontent.com/1469636/89224148-47a6fe00-d5d8-11ea-9712-ae25f6f42b5a.png) |

**Comment by vas3k** [comment]
Давай показывать полноценные. Пока не уверен как лучше, но если чо потом поправим

**Comment by nlopin** [comment]
Заметочка на будущее — надо делать тизеры. В поиске комменты занимают кучу места, а посты торчат в виде заголовков :(

**Comment by nlopin** [comment]
Еще актуально доделать эту задачу? Могу за сегодня и завтра добить её
